CS410_Assignment2/src/SensorApplication/sensor/TemperatureSensorAdapter.java

```diff
@@ -0,0 +1,21 @@
+package sensor;
+
+import sensor.TemperatureSensor;
+
+public class TemperatureSensorAdapter extends SensorAdapter{
+
+	public TemperatureSensorAdapter(){
+		this.sensor = new TemperatureSensor();
+		this.type = sensor.getSensorType();
+		this.value = sensor.senseTemperature();
+		this.status = sensor.getTempReport();
+	}

```

Comment:
Based on the above adapter, it is not possible to create an adapter for a given adaptee. Seems this adapter is always creating a new adaptee inside of itself. It is better to able able to follow the adapter pattern such that we can create an adapter for any given adaptee. for example
public class TemperatureSensorAdapter implements Sensor {
    private TemperatureSensor adaptee;
    public TemperatureSensorAdapter(TemperatureSensor temperatureSensor) {
        this.adaptee = temperatureSensor;
    }
This way we can create a TemperatureSensorAdapter for any TemperatueSensor object we have.